### PR TITLE
chore(controller): prefer using pod failed reason

### PR DESCRIPTION
**Pull request by jialeicui**
## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [ ] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/JobEventHandler.java [hunk 1]
```diff
@@ -224,4 +223,18 @@ private Long getPodStartTime(String jobId) {
...
+
+        var failedReasons = pods.stream().filter(p -> p.getStatus() != null)
+                .filter(p -> Objects.equals(p.getStatus().getPhase(), "Failed"))
+                .map(p -> joinReasons(p.getStatus().getReason(), p.getStatus().getMessage()))
+                .filter(Objects::nonNull).collect(Collectors.toList());
+
+        if (failedReasons.size() == 0) {
+            return null;
+        }
+        return failedReasons.stream().min(String::compareTo).get();
```
Comment (goldenxinxing) [inline server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/JobEventHandler.java:238]: Is it fit to get reason by the first char of failed reason?

Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/JobEventHandler.java:238]: On the native kubelet, the error of the first pod is usually very representative, such as the image does not exist, etc.

If the pod is up, the user can check the cause of the failure directly from the Pod log, which is usually a problem with the user's own program.

File: server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/JobEventHandler.java [hunk 2]
```diff
@@ -182,6 +179,12 @@ private void updateEvalTask(V1Job job, boolean onDelete) {
             }
         }
 
+        // prefer using the pod failed reason, it has more details
+        var podFailedReason = getPodFailedReason(jobName);
+        if (StringUtils.hasText(podFailedReason)) {
+            failedReason = podFailedReason;
```
Comment (anda-ren) [inline server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/JobEventHandler.java:185]: failedReason = f'job failed reason: {failedReason} , pod failed reason: {podFailedReason}' ?

File: server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/JobEventHandler.java (hunk 1 above)
Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/JobEventHandler.java:238]: If there are more than one pod fail, just concat them?

File: server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/JobEventHandler.java (hunk 2 above)
Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/JobEventHandler.java]: All added

File: server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/JobEventHandler.java (hunk 1 above)
Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/JobEventHandler.java]: All added

**Review by anda-ren — APPROVED** [review]
LGTM

**Pull request merged**